src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs

```diff
@@ -60,6 +60,35 @@ public void Dispose()
             ProjectCollection.GlobalProjectCollection.UnloadAllProjects();
         }
 
+        /// <summary>
+        /// Test that if a before.<sln>.targets or after.<sln>.targets file has one of the default targets (Build, Clean, etc.) that it includes only the user-defined target.
+        /// </summary>
+        [Theory]
+        [InlineData("before.MySln.sln.targets")]
+        [InlineData("after.MySln.sln.targets")]
+        [InlineData("name.that.does.Not.Affect.The.Build.targets")]
+        public void SolutionProjectIgnoresDuplicateDefaultTargets(string name)
+        {
+            using (TestEnvironment testEnvironment = TestEnvironment.Create())
+            {
+                TransientTestFolder folder = testEnvironment.CreateFolder(createFolder: true);
+                TransientTestFile sln = testEnvironment.CreateFile(folder, "MySln.sln", @"Microsoft Visual Studio Solution File, Format Version 16.00");
+                TransientTestFile targetsFile = testEnvironment.CreateFile(folder, name,
+                    @"<Project ToolsVersion=""15.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+                        <Target Name=""Build"" AfterTargets=""NonsenseTarget"">
+                        </Target>
+                      </Project>");
+                ProjectInstance[] instances = SolutionProjectGenerator.Generate(SolutionFile.Parse(sln.Path), null, null, _buildEventContext, CreateMockLoggingService());
+                instances.ShouldHaveSingleItem();
+                if (!name.Equals("name.that.does.Not.Affect.The.Build.targets"))
+                {
+                    instances[0].Targets["Build"].AfterTargets.ShouldBe("NonsenseTarget");
```

Comment:
I don't think this is right. When defined in `before`, the dummy target from the bug should be overridden by the generated target "in the body" of the metaproject.

Have you validated that this change behaves correctly using the repro steps from https://github.com/microsoft/msbuild/issues/5071#issuecomment-578863667?

Reply:
Ah, I think this was a design issue. I thought it would be best for the user's version to override the generated version no matter what. I can change it.